packages/framework-provider-aws/src/index.ts

```
@@ -57,63 +9,56 @@ function loadInfrastructurePackage(packageName: string): HasInfrastructure {
-      deleteAllSubscriptions: deleteAllSubscriptions.bind(null, dynamoDB),
-    },
-    // ProviderGraphQLLibrary
-    graphQL: {
-      rawToEnvelope: rawGraphQLRequestToEnvelope,
-      handleResult: requestSucceeded,
-    },
-    // ProviderAPIHandling
-    api: {
-      requestSucceeded,
-      requestFailed,
-    },
-    connections: {
-      storeData: storeConnectionData.bind(null, dynamoDB),
-      fetchData: fetchConnectionData.bind(null, dynamoDB),
-      deleteData: deleteConnectionData.bind(null, dynamoDB),
-      sendMessage: sendMessageToConnection,
-    },
-    // ScheduledCommandsLibrary
-    scheduled: {
-      rawToEnvelope: rawScheduledInputToEnvelope,
-    },
-    // ProviderInfrastructureGetter
-    infrastructure: () => {
-      const infrastructurePackageName = require('../package.json').name + '-infrastructure'
-      let infrastructure: HasInfrastructure | undefined
-      try {
-        infrastructure = loadInfrastructurePackage(infrastructurePackageName)
-      } catch (e) {
-        throw new Error(
-          `The AWS infrastructure package could not be loaded. The following error was thrown: ${e.message}. Please ensure that one of the following actions has been done:\n` +
-            `  - It has been specified in your "devDependencies" section of your "package.json" file. You can do so by running 'npm install --save-dev ${infrastructurePackageName}'\n` +
-            `  - Or it has been installed globally. You can do so by running 'npm install -g ${infrastructurePackageName}'`
-        )
-      }
-
-      return infrastructure.Infrastructure(rockets)
-    },
+  try {
+    require('aws-sdk')
+    const { Provider } = require('./setup')
+    return Provider(rockets)
+  } catch (e) {
+    return {
+      // ProviderEventsLibrary
+      events: {
+        rawToEnvelopes: undefined as any,
+        forEntitySince: undefined as any,
+        latestEntitySnapshot: undefined as any,
```

Not sure about this, but isn't it better to throw an error message?